Test/Unit/Gateway/Response/RefundHandlerTest.php

```diff
@@ -0,0 +1,115 @@
+            ->method('setTransactionId')
+            ->with($provider['lastRefundId']);
+        $paymentMock->expects($this->once())
+            ->method('setTransactionAdditionalInfo')
+            ->with(Transaction::RAW_DETAILS, $provider['lastRefundData']);
+        $paymentMock->expects($this->once())
+            ->method('addTransaction')
+            ->with(TransactionInterface::TYPE_REFUND);
+        $paymentDataMock =  $this->createMock(PaymentDataObject::class);
+        $paymentDataMock->expects($this->once())
+            ->method('getPayment')
+            ->willReturn($paymentMock);
+
+        $handlingSubjectMock['payment'] = $paymentDataMock;
+        $handlingSubjectMock['amount'] = '48';
+
+        $almaPaymentMock =  $this->createMock(AlmaPayment::class);
+        $almaPaymentMock->refunds = $provider['refunds'];
+        $responseMock['almaRefund'] = $almaPaymentMock;
+        $responseMock['isFullRefund'] = $provider['isFullRefund'];
+        $responseMock['customerFee'] = $provider['customerFee'];
+        $refundHandler = $this->createNewRefundHandler();
+        $refundHandler->handle($handlingSubjectMock, $responseMock);
+    }
+
+    public function refundDataProvider(): array
+    {
+        return [
+            'Test not full refund' => [
+                    [
+                        'isFullRefund' => false,
+                        'customerFee' => 0,
+                        'refunds' => $this->getAlmaRefunds(),
+                        'formatPriceReturn' => ['€22', '€48', '€48'],
+                        'lastRefundId' => 'refund_3333333333',
+                        'lastRefundData' => [
+                            'created' => self::CREATED_TIMESTAMP,
+                            'amount' => '€48'
+                        ]
+                    ]
+                ],
+            'Test with full refund' => [
+                    [
+                        'isFullRefund' => true,
+                        'customerFee' => 1600,
+                        'refunds' => $this->getAlmaRefunds(),
+                        'formatPriceReturn' => ['€22', '€48', '€48', '€16', '€48'],
+                        'lastRefundId' => 'refund_3333333333',
+                        'lastRefundData' => [
+                            'created' => self::CREATED_TIMESTAMP,
+                            'amount' => '€48',
+                            'customer_fee' => '€16',
+                            'magento_refund' => '€48'
+                        ]
+                    ]
+                ]
+            ];
+    }
+
+    private function getAlmaRefunds(): array
+    {
+        return [
+            new Refund(['id' => 'refund_1111111111','created' => '1654472700','amount' => '2200']),
+            new Refund(['id' => 'refund_2222222222','created' => '1654472720','amount' => '4800']),
+            new Refund(['id' => 'refund_3333333333','created' => self::CREATED_TIMESTAMP,'amount' => '4800']),
```

Comment:
which difference exists between `CREATED_TIMESTAMP` and the two above ?

it this is just because `CREATED_TIMESTAMP` is used more than once, I suggest you to rename it as `FIXED_TIMESTAMP` (created suggest me that it is a variable timestamp created during the test initialisation)